src/components/MAPI/apps/ClusterDetailApps.tsx

```diff
@@ -0,0 +1,314 @@
+import { useAuthProvider } from 'Auth/MAPI/MapiAuthProvider';
+import UserInstalledApps from 'Cluster/ClusterDetail/UserInstalledApps/UserInstalledApps';
+import { push } from 'connected-react-router';
+import { ingressControllerInstallationURL } from 'lib/docs';
+import { FlashMessage, messageTTL, messageType } from 'lib/flashMessage';
+import { useHttpClientFactory } from 'lib/hooks/useHttpClientFactory';
+import AppDetailsModalMAPI from 'MAPI/apps/AppDetailsModal';
+import { extractErrorMessage } from 'MAPI/organizations/utils';
+import { GenericResponse } from 'model/clients/GenericResponse';
+import * as applicationv1alpha1 from 'model/services/mapi/applicationv1alpha1';
+import * as releasev1alpha1 from 'model/services/mapi/releasev1alpha1';
+import PropTypes from 'prop-types';
+import React, { useLayoutEffect, useMemo, useRef, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { AppConstants } from 'shared/constants';
+import { AppsRoutes } from 'shared/constants/routes';
+import { selectCluster } from 'stores/main/actions';
+import { getProvider } from 'stores/main/selectors';
+import { supportsOptionalIngress } from 'stores/nodepool/utils';
+import styled from 'styled-components';
+import { FlashMessageType } from 'styles';
+import useSWR from 'swr';
+import Button from 'UI/Controls/Button';
+import ClusterDetailPreinstalledApp from 'UI/Display/Cluster/ClusterDetailPreinstalledApp';
+import FlashMessageComponent from 'UI/Display/FlashMessage';
+import NotAvailable from 'UI/Display/NotAvailable';
+
+import ClusterDetailAppLoadingPlaceholder from './ClusterDetailAppLoadingPlaceholder';
+import { filterUserInstalledApps, mapDefaultApps } from './utils';
+
+const LOADING_COMPONENTS = new Array(6).fill(0).map((_, idx) => idx);
+
+function formatAppVersion(appMeta: AppConstants.IAppMetaApp) {
+  const { version } = appMeta;
+
+  // TODO(axbarsan): Handle K8s version EOL dates once available.
```

Comment:
There is no way to get them in the MAPI world yet

Reply:
There is https://github.com/giantswarm/roadmap/issues/336 for that now.